### ISSUE TITLE
Modularization of sagelib: Break out a separate package sagemath-standard-no-symbolics

### ISSUE THREAD

**Issue by mkoeppe** [sagemath/sage#32601]
<div id="comment:0"></div>

This is a subset distribution that includes all of **sagemath-standard** except for `sage.symbolic` and modules that use that.

As preparation, we remove gratuitous module-level imports from `sage.symbolic`, `sage.functions`, `sage.calculus` which block this modularization.
- either by making these imports `lazy_import`
- or by moving module-level imports to individual methods
- or by replacing unnecessary uses of symbolic functions
- or by using .abc (see #32414)
The command `git grep -E '^from.*sage[.](symbolic|calculus|functions).*import'` helps to find these imports.

- #32411/#32413/#32415/#32416: Remove unnecessary uses of `SR` and symbolic functions
- #32599: `sage.structure`, `sage.sets`: Remove hard dependencies on the symbolic ring / polynomial rings
- #32593: `sage.modules`, `sage.matrix`: Remove import-time dependencies on the symbolic ring
- #32708: `sage.tensor.modules`: remove dependency on `sage.manifolds`
- #32712 `sage.tensor.modules`: make doctests involving `SR` optional
- #32717: Remove use of `sage.functions.other.sqrt` (outside of `sage.symbolic`, `sage.functions`, `sage.manifolds`)
- #32731: Avoid `ceil`, `floor` from `sage.functions.other` in non-symbolic uses
- #32794: `sage.functions.log.log`: Move to `sage.misc.functional`
- #33392: `sage.matrix`: Modularization fixes after #33159
- #32724: Replace `... is SR` by `isinstance(..., sage.rings.abc.SymbolicRing)` to handle symbolic subrings
- #32609 Remove more unnecessary uses of `SR` and symbolic functions in `sage.combinat`
- #34215 Removing remaining uses of `is_CallableSymbolicExpression`
- #34217 `sage.combinat.finite_state_machine`: Remove use of "var"
- #25827: Methods `__round__`, `__trunc__`, `__floor__`, `__ceil__`
- #32718: Remove module-level imports from `sage.symbolic` etc. (outside of `sage.symbolic`, `sage.calculus` etc.)
- #32721: Remove abuse of predefined `x` in doctests
- #33673: `sage.categories.classical_crystals`: Remove use of `sage.symbolic`
- #33674: `sage.combinat.root_system.coxeter_type`: Replace module-level import of `SR`

To test:

```
./bootstrap && ./configure && make build && ./sage -sh -c '(cd pkgs/sagemath-standard-no-symbolics && SAGE_NUM_THREADS=16 tox -v -v -v -e sagepython-sagewheels-nopypi)'
```

See also:
- #31695 Modularization of sagelib: Break out separate package **sagemath-symbolics**

Depends on #34547
Depends on #16522
Depends on #34855

CC:  @kliem @egourgoulhon @tobiasdiez @anneschilling

Component: **refactoring**

_Issue created by migration from https://trac.sagemath.org/ticket/32601_

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,2 +1,8 @@
 This is a subset distribution that includes all of `sagemath-standard` except for `sage.symbolic` and modules that use that.
 
+To test:
+
+```
+./bootstrap && make build && ./sage -sh -c '(cd pkgs/sagemath-standard-no-symbolics && tox -v -v -v -e python-sagewheels-nopypi)'
+```
+
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,4 +1,11 @@
 This is a subset distribution that includes all of `sagemath-standard` except for `sage.symbolic` and modules that use that.
+
+As preparation, we remove gratuitous module-level imports from `sage.symbolic`, `sage.functions`, `sage.calculus`. `git grep -E 'sage[.](symbolic|calculus|functions).*import'` helps to find them.
+
+- #32411/#32413/#32415/#32416: Remove unnecessary uses of `SR` and symbolic functions
+- #32599: `sage.structure`, `sage.sets`: Remove hard dependencies on the symbolic ring / polynomial rings
+- #32593: `sage.modules`, `sage.matrix`: Remove import-time dependencies on the symbolic ring
+
 
 To test:
 
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,6 +1,6 @@
 This is a subset distribution that includes all of `sagemath-standard` except for `sage.symbolic` and modules that use that.
 
-As preparation, we remove gratuitous module-level imports from `sage.symbolic`, `sage.functions`, `sage.calculus`. `git grep -E 'sage[.](symbolic|calculus|functions).*import'` helps to find them.
+As preparation, we remove gratuitous module-level imports from `sage.symbolic`, `sage.functions`, `sage.calculus`. The command `git grep -E '^from.*sage[.](symbolic|calculus|functions).*import'` helps to find them.
 
 - #32411/#32413/#32415/#32416: Remove unnecessary uses of `SR` and symbolic functions
 - #32599: `sage.structure`, `sage.sets`: Remove hard dependencies on the symbolic ring / polynomial rings
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -3,6 +3,7 @@
 As preparation, we remove gratuitous module-level imports from `sage.symbolic`, `sage.functions`, `sage.calculus`. The command `git grep -E '^from.*sage[.](symbolic|calculus|functions).*import'` helps to find them.
 
 - #32411/#32413/#32415/#32416: Remove unnecessary uses of `SR` and symbolic functions
+- #32609 Remove more unnecessary uses of `SR` and symbolic functions in `sage.combinat`
 - #32599: `sage.structure`, `sage.sets`: Remove hard dependencies on the symbolic ring / polynomial rings
 - #32593: `sage.modules`, `sage.matrix`: Remove import-time dependencies on the symbolic ring
 
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,6 +1,11 @@
 This is a subset distribution that includes all of `sagemath-standard` except for `sage.symbolic` and modules that use that.
 
-As preparation, we remove gratuitous module-level imports from `sage.symbolic`, `sage.functions`, `sage.calculus`. The command `git grep -E '^from.*sage[.](symbolic|calculus|functions).*import'` helps to find them.
+As preparation, we remove gratuitous module-level imports from `sage.symbolic`, `sage.functions`, `sage.calculus` which block this modularization.
+- either by making these imports `lazy_import`
+- or by moving module-level imports to individual methods
+- or by replacing unnecessary uses of symbolic functions
+- or by using .abc (see #32414)
+The command `git grep -E '^from.*sage[.](symbolic|calculus|functions).*import'` helps to find these imports.
 
 - #32411/#32413/#32415/#32416: Remove unnecessary uses of `SR` and symbolic functions
 - #32609 Remove more unnecessary uses of `SR` and symbolic functions in `sage.combinat`
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -11,7 +11,7 @@
 - #32609 Remove more unnecessary uses of `SR` and symbolic functions in `sage.combinat`
 - #32599: `sage.structure`, `sage.sets`: Remove hard dependencies on the symbolic ring / polynomial rings
 - #32593: `sage.modules`, `sage.matrix`: Remove import-time dependencies on the symbolic ring
-
+- #32708: `sage.tensor.modules`: remove dependency on `sage.manifolds`
 
 To test:
 
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,4 +1,4 @@
-This is a subset distribution that includes all of `sagemath-standard` except for `sage.symbolic` and modules that use that.
+This is a subset distribution that includes all of **sagemath-standard** except for `sage.symbolic` and modules that use that.
 
 As preparation, we remove gratuitous module-level imports from `sage.symbolic`, `sage.functions`, `sage.calculus` which block this modularization.
 - either by making these imports `lazy_import`
@@ -19,3 +19,5 @@
 ./bootstrap && make build && ./sage -sh -c '(cd pkgs/sagemath-standard-no-symbolics && tox -v -v -v -e python-sagewheels-nopypi)'
 ```
 
+See also:
+- #31695 Modularization of sagelib: Break out separate package **sagemath-symbolics**
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -16,7 +16,7 @@
 To test:
 
 ```
-./bootstrap && make build && ./sage -sh -c '(cd pkgs/sagemath-standard-no-symbolics && tox -v -v -v -e python-sagewheels-nopypi)'
+./bootstrap && make build && ./sage -sh -c '(cd pkgs/sagemath-standard-no-symbolics && SAGE_NUM_THREADS=16 tox -v -v -v -e py39-sagewheels-nopypi)'
 ```
 
 See also:
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -12,6 +12,8 @@
 - #32599: `sage.structure`, `sage.sets`: Remove hard dependencies on the symbolic ring / polynomial rings
 - #32593: `sage.modules`, `sage.matrix`: Remove import-time dependencies on the symbolic ring
 - #32708: `sage.tensor.modules`: remove dependency on `sage.manifolds`
+- #32712 `sage.tensor.modules`: make doctests involving `SR` optional
+- #32717: Remove use of `sage.functions.other.sqrt` (outside of `sage.symbolic`, `sage.functions`, `sage.manifolds`)
 
 To test:
 
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -14,6 +14,7 @@
 - #32708: `sage.tensor.modules`: remove dependency on `sage.manifolds`
 - #32712 `sage.tensor.modules`: make doctests involving `SR` optional
 - #32717: Remove use of `sage.functions.other.sqrt` (outside of `sage.symbolic`, `sage.functions`, `sage.manifolds`)
+- #25827: Methods `__round__`, `__trunc__`, `__floor__`, `__ceil__`
 
 To test:
 
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -15,6 +15,7 @@
 - #32712 `sage.tensor.modules`: make doctests involving `SR` optional
 - #32717: Remove use of `sage.functions.other.sqrt` (outside of `sage.symbolic`, `sage.functions`, `sage.manifolds`)
 - #25827: Methods `__round__`, `__trunc__`, `__floor__`, `__ceil__`
+- #32718: Remove module-level imports from `sage.symbolic` etc. (outside of `sage.symbolic`, `sage.calculus` etc.)
 
 To test:
 
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -16,6 +16,7 @@
 - #32717: Remove use of `sage.functions.other.sqrt` (outside of `sage.symbolic`, `sage.functions`, `sage.manifolds`)
 - #25827: Methods `__round__`, `__trunc__`, `__floor__`, `__ceil__`
 - #32718: Remove module-level imports from `sage.symbolic` etc. (outside of `sage.symbolic`, `sage.calculus` etc.)
+- #32721: Remove abuse of predefined `x` in doctests
 
 To test:
 
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -17,6 +17,7 @@
 - #25827: Methods `__round__`, `__trunc__`, `__floor__`, `__ceil__`
 - #32718: Remove module-level imports from `sage.symbolic` etc. (outside of `sage.symbolic`, `sage.calculus` etc.)
 - #32721: Remove abuse of predefined `x` in doctests
+- #32724: Replace `... is SR` by `isinstance(..., sage.rings.abc.SymbolicRing)` to handle symbolic subrings
 
 To test:
 
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -14,6 +14,9 @@
 - #32708: `sage.tensor.modules`: remove dependency on `sage.manifolds`
 - #32712 `sage.tensor.modules`: make doctests involving `SR` optional
 - #32717: Remove use of `sage.functions.other.sqrt` (outside of `sage.symbolic`, `sage.functions`, `sage.manifolds`)
+- #32731: Avoid `ceil`, `floor` from `sage.functions.other` in non-symbolic uses
+- #32794: `sage.functions.log.log`: Move to `sage.misc.functional`
+
 - #25827: Methods `__round__`, `__trunc__`, `__floor__`, `__ceil__`
 - #32718: Remove module-level imports from `sage.symbolic` etc. (outside of `sage.symbolic`, `sage.calculus` etc.)
 - #32721: Remove abuse of predefined `x` in doctests
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -16,7 +16,7 @@
 - #32717: Remove use of `sage.functions.other.sqrt` (outside of `sage.symbolic`, `sage.functions`, `sage.manifolds`)
 - #32731: Avoid `ceil`, `floor` from `sage.functions.other` in non-symbolic uses
 - #32794: `sage.functions.log.log`: Move to `sage.misc.functional`
-
+- #33392: `sage.matrix`: Modularization fixes after #33159
 - #25827: Methods `__round__`, `__trunc__`, `__floor__`, `__ceil__`
 - #32718: Remove module-level imports from `sage.symbolic` etc. (outside of `sage.symbolic`, `sage.calculus` etc.)
 - #32721: Remove abuse of predefined `x` in doctests
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -8,7 +8,6 @@
 The command `git grep -E '^from.*sage[.](symbolic|calculus|functions).*import'` helps to find these imports.
 
 - #32411/#32413/#32415/#32416: Remove unnecessary uses of `SR` and symbolic functions
-- #32609 Remove more unnecessary uses of `SR` and symbolic functions in `sage.combinat`
 - #32599: `sage.structure`, `sage.sets`: Remove hard dependencies on the symbolic ring / polynomial rings
 - #32593: `sage.modules`, `sage.matrix`: Remove import-time dependencies on the symbolic ring
 - #32708: `sage.tensor.modules`: remove dependency on `sage.manifolds`
@@ -17,10 +16,11 @@
 - #32731: Avoid `ceil`, `floor` from `sage.functions.other` in non-symbolic uses
 - #32794: `sage.functions.log.log`: Move to `sage.misc.functional`
 - #33392: `sage.matrix`: Modularization fixes after #33159
+- #32724: Replace `... is SR` by `isinstance(..., sage.rings.abc.SymbolicRing)` to handle symbolic subrings
+- #32609 Remove more unnecessary uses of `SR` and symbolic functions in `sage.combinat`
 - #25827: Methods `__round__`, `__trunc__`, `__floor__`, `__ceil__`
 - #32718: Remove module-level imports from `sage.symbolic` etc. (outside of `sage.symbolic`, `sage.calculus` etc.)
 - #32721: Remove abuse of predefined `x` in doctests
-- #32724: Replace `... is SR` by `isinstance(..., sage.rings.abc.SymbolicRing)` to handle symbolic subrings
 
 To test:
 
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -25,7 +25,7 @@
 To test:
 
 ```
-./bootstrap && make build && ./sage -sh -c '(cd pkgs/sagemath-standard-no-symbolics && SAGE_NUM_THREADS=16 tox -v -v -v -e py39-sagewheels-nopypi)'
+./bootstrap && ./configure && make build && ./sage -sh -c '(cd pkgs/sagemath-standard-no-symbolics && SAGE_NUM_THREADS=16 tox -v -v -v -e py39-sagewheels-nopypi)'
 ```
 
 See also:
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -21,6 +21,7 @@
 - #25827: Methods `__round__`, `__trunc__`, `__floor__`, `__ceil__`
 - #32718: Remove module-level imports from `sage.symbolic` etc. (outside of `sage.symbolic`, `sage.calculus` etc.)
 - #32721: Remove abuse of predefined `x` in doctests
+- #33673: `sage.categories.classical_crystals`: Remove use of `sage.symbolic`
 
 To test:
 
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -22,6 +22,7 @@
 - #32718: Remove module-level imports from `sage.symbolic` etc. (outside of `sage.symbolic`, `sage.calculus` etc.)
 - #32721: Remove abuse of predefined `x` in doctests
 - #33673: `sage.categories.classical_crystals`: Remove use of `sage.symbolic`
+- #33674: `sage.combinat.root_system.coxeter_type`: Replace module-level import of `SR`
 
 To test:
 
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -27,7 +27,7 @@
 To test:
 
 ```
-./bootstrap && ./configure && make build && ./sage -sh -c '(cd pkgs/sagemath-standard-no-symbolics && SAGE_NUM_THREADS=16 tox -v -v -v -e py39-sagewheels-nopypi)'
+./bootstrap && ./configure && make build && ./sage -sh -c '(cd pkgs/sagemath-standard-no-symbolics && SAGE_NUM_THREADS=16 tox -v -v -v -e sagepython-sagewheels-nopypi)'
 ```
 
 See also:
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -18,6 +18,8 @@
 - #33392: `sage.matrix`: Modularization fixes after #33159
 - #32724: Replace `... is SR` by `isinstance(..., sage.rings.abc.SymbolicRing)` to handle symbolic subrings
 - #32609 Remove more unnecessary uses of `SR` and symbolic functions in `sage.combinat`
+- #34215 Removing remaining uses of `is_CallableSymbolicExpression`
+- #34217 `sage.combinat.finite_state_machine`: Remove use of "var"
 - #25827: Methods `__round__`, `__trunc__`, `__floor__`, `__ceil__`
 - #32718: Remove module-level imports from `sage.symbolic` etc. (outside of `sage.symbolic`, `sage.calculus` etc.)
 - #32721: Remove abuse of predefined `x` in doctests
``````

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`bb844e6`](https://github.com/sagemath/sagetrac-mirror/commit/bb844e670bddf83d9995cf8f4fc4465433915453)** to **[`d5f9cf5`](https://github.com/sagemath/sagetrac-mirror/commit/d5f9cf5a452f623a61bd16d3f699e05732fb63f0)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:98"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/d5f9cf5a452f623a61bd16d3f699e05732fb63f0"><code>d5f9cf5</code></a></td><td><code>Merge tag '9.8.beta0' into t/32601/modularization_of_sagelib__break_out_a_separate_package_sagemath_standard_no_symbolics</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:99"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/9de7a1afcee75918f910462b01ed1cad348f577e"><code>9de7a1a</code></a></td><td><code>Merge tag '9.8.beta3' into t/32601/modularization_of_sagelib__break_out_a_separate_package_sagemath_standard_no_symbolics</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`d5f9cf5`](https://github.com/sagemath/sagetrac-mirror/commit/d5f9cf5a452f623a61bd16d3f699e05732fb63f0)** to **[`9de7a1a`](https://github.com/sagemath/sagetrac-mirror/commit/9de7a1afcee75918f910462b01ed1cad348f577e)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`9de7a1a`](https://github.com/sagemath/sagetrac-mirror/commit/9de7a1afcee75918f910462b01ed1cad348f577e)** to **[`c4425da`](https://github.com/sagemath/sagetrac-mirror/commit/c4425dae94c3b95e0d2eb3b389e9df43cd5f5815)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:100"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/c4425dae94c3b95e0d2eb3b389e9df43cd5f5815"><code>c4425da</code></a></td><td><code>Merge tag '9.8.beta4' into t/32601/modularization_of_sagelib__break_out_a_separate_package_sagemath_standard_no_symbolics</code></td></tr></table>

**Comment by mkoeppe** [comment]
Changed dependencies from **#32913** to none

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:102"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/b102b122f60e841af79fd89942f457e9399877c6"><code>b102b12</code></a></td><td><code>pkgs/sagemath-standard-no-symbolics: Update</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`c4425da`](https://github.com/sagemath/sagetrac-mirror/commit/c4425dae94c3b95e0d2eb3b389e9df43cd5f5815)** to **[`b102b12`](https://github.com/sagemath/sagetrac-mirror/commit/b102b122f60e841af79fd89942f457e9399877c6)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`b102b12`](https://github.com/sagemath/sagetrac-mirror/commit/b102b122f60e841af79fd89942f457e9399877c6)** to **[`cf6b1fb`](https://github.com/sagemath/sagetrac-mirror/commit/cf6b1fbab7b2e24281cc5a6a81fc4586eecdefd7)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:103"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/b82d66057963f054dcee51c43f1c2f7c67809d73"><code>b82d660</code></a></td><td><code>Merge tag '9.8.beta5' into t/32601/modularization_of_sagelib__break_out_a_separate_package_sagemath_standard_no_symbolics</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/cf6b1fbab7b2e24281cc5a6a81fc4586eecdefd7"><code>cf6b1fb</code></a></td><td><code>Fix imports</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:104"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/0810d7a77c1f5384c3d92ffad718239b98e2975d"><code>0810d7a</code></a></td><td><code>Fix imports</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/2d7dd5393816cf213c5ecb7e9acf8f8bb71d761e"><code>2d7dd53</code></a></td><td><code>src/sage/all_cmdline.py: Do not use lazy_import for predefined x</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`cf6b1fb`](https://github.com/sagemath/sagetrac-mirror/commit/cf6b1fbab7b2e24281cc5a6a81fc4586eecdefd7)** to **[`2d7dd53`](https://github.com/sagemath/sagetrac-mirror/commit/2d7dd5393816cf213c5ecb7e9acf8f8bb71d761e)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:105"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/bd534cb80b3860d955c57c92aa86d7923d2c4bc0"><code>bd534cb</code></a></td><td><code>pkgs/sagemath-standard-no-symbolics/sage_setup: Remove</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`2d7dd53`](https://github.com/sagemath/sagetrac-mirror/commit/2d7dd5393816cf213c5ecb7e9acf8f8bb71d761e)** to **[`bd534cb`](https://github.com/sagemath/sagetrac-mirror/commit/bd534cb80b3860d955c57c92aa86d7923d2c4bc0)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`bd534cb`](https://github.com/sagemath/sagetrac-mirror/commit/bd534cb80b3860d955c57c92aa86d7923d2c4bc0)** to **[`6b1c674`](https://github.com/sagemath/sagetrac-mirror/commit/6b1c6747ce6911ea824d9aea2c6cb2339b7d0d21)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:106"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/6b1c6747ce6911ea824d9aea2c6cb2339b7d0d21"><code>6b1c674</code></a></td><td><code>Fix imports</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:107"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/e426400637388e8d35073c2d6b3e4330cdb32df2"><code>e426400</code></a></td><td><code>Fix imports</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`6b1c674`](https://github.com/sagemath/sagetrac-mirror/commit/6b1c6747ce6911ea824d9aea2c6cb2339b7d0d21)** to **[`e426400`](https://github.com/sagemath/sagetrac-mirror/commit/e426400637388e8d35073c2d6b3e4330cdb32df2)**

**Comment by mkoeppe** [comment]
Dependencies: **#34547, #16522**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:109"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/250da03c8602100d6df6b405e23ded5644bdc217"><code>250da03</code></a></td><td><code>trac 34547: use lazy imports in much of sage.interfaces.all</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/9566546fec29409bb5d700275495720b224cfa01"><code>9566546</code></a></td><td><code>trac 34547: remove two "import sage.interfaces.all" lines</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/5789608df64b0eee6ae7332b2dbb04d2d1da3fc4"><code>5789608</code></a></td><td><code>Merge #34547</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/a29a25d997a95db0d51aaef371c806791082658b"><code>a29a25d</code></a></td><td><code>trac 16522: introduce debugging routine to access LazyImport attributes and a routine to clean namespaces of misconfigured LazyImport objects</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/1fa627b3ff2b67a12fac555a8b31efb84f7d0ab8"><code>1fa627b</code></a></td><td><code>clean sage.all and user_globals upon initialization</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/6e384f50ecd6ff8fa15cdc23f370fcd8c0683c61"><code>6e384f5</code></a></td><td><code>src/sage/misc/lazy_import.pyx: Codestyle/markup fixes</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/4acb9d7fabd2c0c81487c0eaf8db370dfa8a082c"><code>4acb9d7</code></a></td><td><code>src/sage/misc/lazy_import.pyx (clean_namespace): Default for namespace like lazy_import</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/05ea54e11921f78b9159c90c50f14e011e7eb433"><code>05ea54e</code></a></td><td><code>src/sage/misc/lazy_import.pyx: Codestyle fixes</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/9837dec14225aec881004e98ed92c2a61b920906"><code>9837dec</code></a></td><td><code>Change order of cases in `Partitions` to put cheaper tests first</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/7f1da655e8195f7442e108af5df9b7ba23414789"><code>7f1da65</code></a></td><td><code>Merge #16522</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`e426400`](https://github.com/sagemath/sagetrac-mirror/commit/e426400637388e8d35073c2d6b3e4330cdb32df2)** to **[`7f1da65`](https://github.com/sagemath/sagetrac-mirror/commit/7f1da655e8195f7442e108af5df9b7ba23414789)**

**Comment by mkoeppe** [comment]
Changed dependencies from **#34547, #16522** to **#34547, #16522, #34855**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:111"></div>

Branch pushed to git repo; I updated commit sha1. Last 10 new commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/2f4b0539a17cf2dd9ff53ccb85a3b7f8f218e6a8"><code>2f4b053</code></a></td><td><code>pkgs/sagemath-standard/setup.py: Run finder also for sdist, egg_info, dist_info</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/00deebede6ae54768e8648bee89038e26d3515f8"><code>00deebe</code></a></td><td><code>pkgs/sagemath-objects/setup.py: Run finder also for sdist, egg_info, dist_info</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/39bd748020fb651e5aeb7640ffe182e493353c79"><code>39bd748</code></a></td><td><code>src/sage/dynamics/arithmetic_dynamics/projective_ds.py: Remove unused import of typing_extensions</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/e22943612656b225396e4dea4fae654108b3cf3d"><code>e229436</code></a></td><td><code>Merge #34859</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/2b39b3d2e2b0a8720dfcd764345c03c120e06396"><code>2b39b3d</code></a></td><td><code>Merge #34858</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/c59622a42624037d4eba61e353ea07ae00bd55f6"><code>c59622a</code></a></td><td><code>build/pkgs/sagemath*/spkg-src: Use build</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/7fc93d06de9f0e34d2ea2077f1a022750c722235"><code>7fc93d0</code></a></td><td><code>build/pkgs/sagemath*/spkg-src: Use --skip-dependency-check</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/4940db4cf4d32617f23a87e04bddaa01061a7f5f"><code>4940db4</code></a></td><td><code>build/make/Makefile.in: Add targets SPKG-src for script packages</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/573fcfdfba04041260f8e06e54e37eb8a9f85da4"><code>573fcfd</code></a></td><td><code>build/make/Makefile.in: Move targets pypi-sdists, wheel, pypi-wheels here; use SPKG-sdist targets</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/8039865f58a8786492b6bebcd3ae0eb3d036a8d0"><code>8039865</code></a></td><td><code>Merge #34855</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`7f1da65`](https://github.com/sagemath/sagetrac-mirror/commit/7f1da655e8195f7442e108af5df9b7ba23414789)** to **[`8039865`](https://github.com/sagemath/sagetrac-mirror/commit/8039865f58a8786492b6bebcd3ae0eb3d036a8d0)**

**Comment by mkoeppe** [comment]
Removed branch from issue description; replaced by PR #35150

**Comment by orlitzky** [comment]
This is a WONTFIX for upstream Sage, please use https://github.com/passagemath/passagemath if you need these distributions.
